forge-gui/res/cardsfolder/upcoming/ambulatory_edifice.txt

```diff
@@ -0,0 +1,7 @@
+Name:Ambulatory Edifice
+ManaCost:2 B
+Types:Artifact Creature Phyrexian Construct
+PT:3/2
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDebuff | TriggerDescription$ When CARDNAME enters the battlefield, you may pay 2 life. When you do, target creature gets -1/-1 until end of turn.
+SVar:TrigDebuff:AB$ Pump | Cost$ PayLife<2> | Optional$ True | ValidTgts$ Creature | TgtPrompt$ Select target creature | NumAtt$ -1 | NumDef$ -1 | IsCurse$ True
```

Comment:
reflexive trigger not handled

Reply:
not sure what you mean by this

Reply:
you can use the rules to look it up:
https://yawgatog.com/resources/magic-rules/#R60312

Reply:
@squee1968 we use ImmediateTrigger for this

Reply:
good thing you explained that. yawgatog was a little too confusing for this early in the morning :-}

Reply:
you can't just change the effect alone to make it work
please look at the card script that is referenced as an example to see how it should be used

Reply:
no, checkout Tolarian Kraken: https://github.com/Card-Forge/forge/blob/256e6c88943b0f01ba1bf78423f1c9daf121b84b/forge-gui/res/cardsfolder/t/tolarian_kraken.txt#L5-L7

* The Trigger has the Cost (2. line)
* and the Effect is on the next one (3. line)